src/lockfile/types.ts

```diff
@@ -15,3 +15,29 @@ export type ParsedLock = {
 export interface PnpmLockFile {
   packages: { [name: string]: any };
 }
+
+export interface NpmWorspacesInfo {
+  version: "string";
+  workspaces: Record<"packages", string[]>;
+}
+
+export interface NpmSynlinkInfo {
+  resolved: string; // Where the package is  reslved from.
+  link: boolean; // A flag to indicate that this is a symbolic link.
+  integrity?: 'sha512' | 'sha1';
+  dev?: boolean;
+  optional?: boolean;
+  devOptional?: boolean;
+  dependendcies?: { [key in string]: LockDependency; }
+}
+
+export interface NpmLockFile {
+  name: string;
+  version: string;
+  lockfileVersion?: number; // 1: v5, v6; 2: backwards compatible v7; 3: non-backwards compatible v7 
+  requires?: boolean;
+  packages?: {
+    ""?: NpmWorspacesInfo, // Monorepo root
```

Comment:
Is this key actually a literal empty string?

Reply:
The key with an empty string is the monorepo root.

> The root project is typically listed with a key of "", and all other packages are listed with their relative paths from the root project folder.